src/Tizen.NUI.WindowSystem/src/public/SoftkeyService.cs

```diff
@@ -0,0 +1,292 @@
+﻿/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.NUI.WindowSystem.Shell
+{
+    /// <summary>
+    /// Class for the Tizen softkey service.
+    /// </summary>
+    /// This class is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class SoftkeyService : IDisposable
+    {
+        private TizenShell _tzsh;
+        private IntPtr _softkeyService;
+        private int _tzshWin;
+        private bool disposed = false;
+        private bool isDisposeQueued = false;
+
+        private Interop.SoftkeyService.SoftkeyVisibleEventCallback _onVisibleChanged;
+        private Interop.SoftkeyService.SoftkeyExpandEventCallback _onExpandChanged;
+        private Interop.SoftkeyService.SoftkeyOpacityEventCallback _onOpacityChanged;
+        
+        private event EventHandler<VisibleState> _visibleChanged;        
+        private event EventHandler<ExpandState> _expandChanged;        
+        private event EventHandler<OpacityState> _opacityChanged;
+
+        /// <summary>
+        /// Creates a new Softkey Service handle.
+        /// </summary>
+        /// <param name="tzShell">The TizenShell instance.</param>
+        /// <param name="win">The window to provide service of the quickpanel.</param>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
```

Comment:
You don't need to throw OOM exception. In case of OOM, dotnet runtime will be stopped first.

Reply:
You mean that, if OOM error is occurred, dotnet runtime will be stopped first, so in Tizen.NUI.WindowSystem module have not to control about OOM error?

Reply:
Yes, WindowSystem module can't control it.

Reply:
I removed all of OOM exceptions